core/ecschema-editing/src/Editing/ECClasses.ts

```diff
@@ -0,0 +1,262 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Editing
+ */
+
+import {
+  ECClass, ECObjectsError, ECObjectsStatus, Enumeration, EnumerationPropertyProps, PrimitiveArrayPropertyProps,
+  PrimitivePropertyProps, PrimitiveType, SchemaItemKey, SchemaItemType, SchemaMatchType, StructArrayPropertyProps,
+  StructClass, StructPropertyProps,
+} from "@bentley/ecschema-metadata";
+import { PropertyEditResults, SchemaContextEditor  } from "./Editor";
+import { MutableClass } from "./Mutable/MutableClass";
+
+/**
+ * @alpha
+ * Acts as a base class for schema class creation. Enables property creation.
+ */
+export class ECClasses {
+  protected constructor(protected _schemaEditor: SchemaContextEditor) { }
+
+  /**
+   * Create a primitive property on class identified by the given SchemaItemKey.
+   * @param classKey The SchemaItemKey of the class.
+   * @param name The name of the new property.
+   * @param type The PrimitiveType assigned to the new property.
+   */
+  public async createPrimitiveProperty(classKey: SchemaItemKey, name: string, type: PrimitiveType): Promise<PropertyEditResults> {
+    const schema = (await this._schemaEditor.schemaContext.getCachedSchema(classKey.schemaKey, SchemaMatchType.Latest));
```

Comment:
So many of these functions differ by only one or a handful of lines of code. Consolidate?

Reply:
done